app/src/main/java/org/simple/clinic/instantsearch/InstantSearchUpdate.kt

```diff
@@ -43,6 +43,8 @@ class InstantSearchUpdate @Inject constructor(
       } else {
         searchResultClicked(model, event.patientId)
       }
+      is PatientAlreadyHasAnExistingNHID -> dispatch(ShowNHIDErrorDialog)
+      is PatientDoesNotHaveAnExistingNHID -> dispatch(OpenLinkIdWithPatientScreen(event.patientId, model.additionalIdentifier!!))
```

Comment:
The handling of this change should be a separate commit and not part of this one. What we should do is use `noChange()` to let the tests pass and then handle this in another commit.